server/src/main/java/com/defold/extender/ExtenderController.java

```diff
@@ -365,7 +365,10 @@ public void buildEngineAsync(HttpServletRequest _request,
                 }
             }
             else {
-                LOGGER.info("Keeping job folder due to debug flags");
+                if (!deleteDirectory)
+                {
+                    LOGGER.info("Keeping job folder due to debug flags");
+                }
```

Comment:
Since it's an asynchronous operation, I think keeping the job folder is implied.

Also, the message was about the debug variables, and behavior should reflect that.